### ISSUE TITLE
Route the internet traffic to a specific client

### ISSUE THREAD

**Issue by andycandy-de** [ngoduykhanh/wireguard-ui#300]
Hello,
I want to route all the internet traffic in my wireguard setup to a specific client

```
P1 - S - P0 - LAN - internet
Pn /
```

I am not sure if I understand the allowed IPs configuration. I tried to configure it that all traffic will be routed to P0

```
S
IP:  10.252.1.1/16
P0 Allowed IPs 0.0.0.0/0
PX Allowed IPs 10.252.2.X/32 (e.g. 10.252.2.1/32)

P0
IP: 10.252.1.2/16
Allowed IPs 10.252.2.0/24

PX
IP: 10.252.2.X/16 (e.g. 10.252.2.1/16)
Allowed IPs 0.0.0.0/0
```

But this doesn't work. What's wrong with this? I hope you understand my ideas and you can help me to archive my goal.

**Comment by andycandy-de** [comment]
My `wg0.conf` looks like this

```
[Interface]
Address = 10.252.0.0/16
ListenPort = 51820
PrivateKey = SECRET
MTU = 1450
PostUp =
PostDown =


# Name:         P01
[Peer]
PublicKey = SECRET
PresharedKey = SECRET
AllowedIPs = 10.252.2.1/32

# Name:         P00
[Peer]
PublicKey = SECRET
PresharedKey = SECRET
AllowedIPs = 10.252.1.1/32,0.0.0.0/5,8.0.0.0/7,10.0.0.0/9,10.128.0.0/10,10.192.0.0/11,10.224.0.0/12,10.240.0.0/13,10.248.0.0/14,10.253.0.0/16,10.254.0.0/15,11.0.0.0/8,12.0.0.0/6,16.0.0.0/4,32.0.0.0/3,64.0.0.0/2,128.0.0.0/1

# Name:         P02
[Peer]
PublicKey = SECRET
PresharedKey = SECRET
AllowedIPs = 10.252.2.2/32
```

With this config only `P00` can connect the server. Other clients (e.g. P01) cannot complete the handshake. When I change the config of `P00` to ...

```
# Name:         P00
[Peer]
PublicKey = SECRET
PresharedKey = SECRET
AllowedIPs = 10.252.1.1/32
```

... all clients can connect the server. But the traffic should be routed to `P00`. What can I do? I have no idea why my configuration doesn't work.

**Comment by andycandy-de** [comment]
OK, I found the issue
Allowed IPs of `P00` includes the device IP of `P01`. I tested to exclude the ip from Allowed IPs and it works
the routing works like expected
bad thing is that that's not what I want... It should work with any ip on `P01`
I am not sure, how I can solve it

**Comment by andycandy-de** [comment]
Solution for my issue is to set following things in the PostUp sciprt

```
iptables -t mangle -A PREROUTING -i wg0 -j MARK --set-mark 0x9d9
ip rule add fwmark 0x9d9 table 2468
ip rule add not fwmark 0x9d9 table main
```

And to add `Table = 2468` in the `wg0.conf` file. Sadly the Table setting is not configurable via the UI so it will be overwritten each config change. I added the possibility to set the Table via UI with this Pull Request https://github.com/ngoduykhanh/wireguard-ui/pull/308

**Comment by andycandy-de** [comment]
@ngoduykhanh This issue is solved with the pull request Pull Request #308

**Comment by andycandy-de** [comment]
@ngoduykhanh This issue is solved with the pull request Pull Request https://github.com/ngoduykhanh/wireguard-ui/pull/308
